src/pages/ReportSettingsPage.js

```diff
@@ -73,61 +66,47 @@ class ReportSettingsPage extends Component {
     }
 
     /**
-     * When the user dismisses the error updating the policy room name,
-     * reset the report name to the previously saved name and clear errors.
+     * @param {Object} values - form input values passed by the Form component
      */
-    resetToPreviousName() {
-        this.setState({newRoomName: this.props.report.reportName});
-        Report.clearPolicyRoomNameErrors(this.props.report.reportID);
-    }
+    validateAndUpdatePolicyRoomName(values) {
+        Keyboard.dismiss();
 
-    validateAndUpdatePolicyRoomName() {
-        if (!this.validate()) {
+        // When the room name has not changed, skip the Form submission
+        if (values.newRoomName === this.props.report.reportName) {
             return;
         }
-        Keyboard.dismiss();
-        Report.updatePolicyRoomName(this.props.report, this.state.newRoomName);
+        Report.updatePolicyRoomName(this.props.report, values.newRoomName);
     }
 
-    validate() {
+    /**
+     * @param {Object} values - form input values passed by the Form component
+     * @returns {Boolean}
+     */
+    validate(values) {
         const errors = {};
 
         // When the report name is not changed, skip the form submission. Added check here to keep the code clean
```

Comment:
I think we can get rid of this comment since it's related to the code block that was deleted

Reply:
sure